src/main/java/com/sonymobile/tools/gerrit/gerritevents/dto/events/ChangeBasedEvent.java

```diff
@@ -75,49 +88,19 @@ public void setChange(Change change) {
         this.change = change;
     }
 
-/**
-     * The changed files in this patchset.
-     */
-    private List<String> files;
-
-
-
     /**
      * Queries gerrit for the files included in this patch set.
      *
      * @param gerritQueryHandler the query handler, responsible for the queries to gerrit.
      * @return a list of files that are part of this patch set.
+     *
+     * @deprecated use {@link Change#getFiles(GerritQueryHandler)} instead.
      */
+    @Deprecated
     public List<String> getFiles(GerritQueryHandler gerritQueryHandler) {
-        if (files == null) {
-            files = new LinkedList<String>();
-            try {
-                List<JSONObject> jsonList = gerritQueryHandler.queryFiles("change:" + getChange().getId());
-                for (JSONObject json : jsonList) {
-                    if (json.has("type") && "stats".equalsIgnoreCase(json.getString("type"))) {
-                        continue;
-                    }
-                    if (json.has("currentPatchSet")) {
-                        JSONObject currentPatchSet = json.getJSONObject("currentPatchSet");
-                        if (currentPatchSet.has("files")) {
-                            JSONArray changedFiles = currentPatchSet.optJSONArray("files");
-                            for (int i = 0; i < changedFiles.size(); i++) {
-                                JSONObject file = changedFiles.getJSONObject(i);
-                                files.add(file.getString("file"));
-                            }
-                        }
-                    }
-                }
-            } catch (IOException e) {
-                logger.error("IOException occured. ", e);
-            } catch (GerritQueryException e) {
-                logger.error("Bad query. ", e);
-            }
-        }
-        return files;
+        return change.getFiles(gerritQueryHandler);
```

Comment:
Since the old behavior has changed in `FileHelper.getFilesByChange` (it now returns `null` instead of an empty list), just to be safe I think we need to preserve the old behavior here. E.g. `if returned == null { return Collections.emptyList() }`